R-packages/covidcast/R/cor.R

```diff
@@ -21,46 +20,53 @@
 #' @param use,method Arguments to pass to `cor()`, with "na.or.complete" the
 #'   default for `use` (different than `cor()`) and "pearson" the default for
 #'   `method` (same as `cor()`).
-#' 
+#'
 #' @return A data frame with first column `geo_value` or `time_value` (depending
 #'   on `by`), and second column `value`, which gives the correlation.
 #'
 #' @importFrom stats cor
 #' @export
-covidcast_cor = function(x, y, dt_x = 0, dt_y = 0,
+covidcast_cor <- function(x, y, dt_x = 0, dt_y = 0,
                          by = c("geo_value", "time_value"),
-                         use = "na.or.complete", 
+                         use = "na.or.complete",
                          method = c("pearson", "kendall", "spearman")) {
-  x = latest_issue(x)
-  y = latest_issue(y)
-  by = match.arg(by)
-  method = match.arg(method)
+  x <- latest_issue(x)
+  y <- latest_issue(y)
+  by <- match.arg(by)
+  method <- match.arg(method)
 
   # Join the two data frames together by pairs of geo_value and time_value
-  z = dplyr::full_join(x, y, by = c("geo_value", "time_value"))
+  z <- dplyr::full_join(x, y, by = c("geo_value", "time_value"))
 
   # Make sure that we have a complete record of dates for each geo_value (fill
   # with NAs as necessary)
-  z_all = z %>% dplyr::group_by(geo_value) %>%
-    dplyr::summarize(time_value = seq.Date(as.Date(min(time_value)),
-                                           as.Date(max(time_value)),
-                                           by = "day")) %>%
-    dplyr::ungroup()
-  z = dplyr::full_join(z, z_all, by = c("geo_value", "time_value"))
+  z_all <- dplyr::group_by(z, .data$geo_value)
+  z_all <- dplyr::summarize(z_all,
+                            time_value = seq.Date(
+                              as.Date(min(.data$time_value)),
+                              as.Date(max(.data$time_value)),
+                              by = "day")
+                            )
+  z_all <- dplyr::ungroup(z_all)
+
+  z <- dplyr::full_join(z, z_all, by = c("geo_value", "time_value"))
 
   # Perform time shifts, then compute appropriate correlations and return
-  return(z %>% dplyr::group_by(geo_value) %>% # group by geo value
-         dplyr::arrange(time_value) %>%  # sort rows by increasing time
-         dplyr::mutate(value.x = shift(value.x, n = dt_x), # shift values
-                       value.y = shift(value.y, n = dt_y)) %>%
-         dplyr::ungroup() %>% # get rid of grouping by geo value
-         dplyr::group_by(dplyr::across(dplyr::all_of(by))) %>% # group by what's specified
-         dplyr::summarize(value = cor(x = value.x, y = value.y, # compute cor
-                                      use = use, method = method)))
+  z <- dplyr::group_by(z, .data$geo_value)
```

Comment:
Does the warning require that we unchain these lines?  I find it to be much more understandable with the chaining.

Reply:
I just generally tried to avoid pipes in package because they can be hard to debug (the traceback gives the entire pipe), particularly when users are submitting bugs that may be hard to replicate without all of their data. I did not have time to do it everyone in the package, but did here when I was fixing the other warning.